test/test.dart

```diff
@@ -545,23 +542,37 @@ Map<String, String?> parseTestArguments(List<String>? arguments) {
 
 extension on Map<String, String?> {
   void addArgument(ArgResults parsedResult, String argName) {
-    final value = parsedResult.wasParsed(argName) ? parsedResult[argName]?.toString() : Platform.environment[argName];
+    late String? value;
+    if (parsedResult.wasParsed(argName)) {
+      if (parsedResult[argName] != null && parsedResult[argName] != "null") {
```

Comment:
```suggestion
      if (parsedResult[argName] != null) {
```
No need to check for `"null"`. I actually preferred it as it was.

Reply:
I believe this was fixed before you did a change. I can revert of course